### PR TITLE
5 space availability dates

### PR DESCRIPTION
**Pull request by junyuanxue**
Rendered calendar in Sinatra. Stored dates for booking requests in Database using ajax requests.
(Note: currently dates are stored within bookings table, however should eventually be stored within requests(?) table. Once confirmed booking, pushed into bookings table.)

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -22,6 +24,37 @@ class MakersBnb < Sinatra::Base
...
   end
 
+  get '/reservation' do
+    File.read(File.join('public', 'calendar.html'))
+  end
+
+  post '/reservation' do
+    Request.create(start_date: params[:start_date], end_date: params[:end_date], status: 1)
+    p params[:start_date]
+    p params[:end_date]
```
Comment (Managram) [inline app/app.rb]: you can get rid of the p

File: app/models/booking.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+
+class Booking
+
+	include DataMapper::Resource
+
+	property :id, Serial
```
Comment (Managram) [inline app/models/booking.rb]: ALIGNMENT

File: spec/helpers.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    fill_in :price,       with: space.price
+    click_button("create space")
+  end
+
+  # def space_request(request)
+  #   visit '/request/new'
+  #   fill_in :start_date, with: request.start_date
+  #   fill_in :end_date,   with: request.end_date
+  #   click_button("send request")
+  # end
```
Comment (Managram) [inline spec/helpers.rb]: unless you need this delete it

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -8,16 +8,38 @@
...
 
+  config.include FactoryGirl::Syntax::Methods
   config.include Capybara::DSL
+  config.include Helpers
+
   config.before(:suite) do
     DatabaseCleaner.strategy = :transaction
     DatabaseCleaner.clean_with(:truncation)
+    FactoryGirl.lint
+    # FactoryGirl.find_definitions
```
Comment (Managram) [inline spec/spec_helper.rb]: here too delete unless necessary

File: app/models/booking.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  property :start_date, Date
+  property :end_date, Date
+
+end
+
+DataMapper.setup(:default, "postgres://localhost/makersbnb_#{ENV['RACK_ENV']}")
```
Comment (rhiannonruth) [inline app/models/booking.rb:14]: these lines are data_mapper_setup

**Comment by Managram** [comment]
:shipit:

**Pull request merged**